File: python/quick-sort.py
import sys

def quick_sort(arr):
    if not arr: 
        return []
    pivot = arr[0]
    lower = [x for x in arr[1:] if x < pivot]
    upper = [x for x in arr[1:] if x >= pivot]
    return quick_sort(lower) + [pivot] + quick_sort(upper)
    
arr = list(map(int, sys.argv[1].split(',')))
print(quick_sort(arr))

File: scala/quick-sort.scala
object Main extends App {
    
  def quicksort(arr: List[Int]): List[Int] = arr match {
    case Nil => Nil
    case pivot :: tail =>
      val (lower, upper) = tail.partition(_ < pivot)
      quicksort(lower) ::: pivot :: quicksort(upper)
  }
  
  println(quicksort((args(0).split(",").map(_.toInt).toList)).mkString(", "))
}

